chain/jsonrpc-primitives/src/errors.rs

```diff
@@ -14,12 +14,33 @@ pub struct RpcParseError(pub String);
 #[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
 #[serde(deny_unknown_fields)]
 pub struct RpcError {
+    #[serde(flatten)]
+    pub error_struct: Option<RpcErrorKind>,
+    /// Deprecated please use the `error_struct` instead
     pub code: i64,
+    /// Deprecated please use the `error_struct` instead
     pub message: String,
+    /// Deprecated please use the `error_struct` instead
     #[serde(skip_serializing_if = "Option::is_none")]
     pub data: Option<Value>,
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
+#[serde(tag = "name", content = "cause", rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum RpcErrorKind {
+    ValidationError(ValidationErrorKind),
+    HandlerError(Value),
+    InternalError(Value),
```

Comment:
Well, now we have another problem, our structure falls apart and we will have:

```json
{
  "name": "INTERNAL_ERROR",
  "cause": {
    "error_message": "..."
  }
}
```

We probably want:

```json
{
  "name": "INTERNAL_ERROR",
  "cause": {
    "name": "???",
    "info": {
      "error_message": "..."
    }
  }
}
```

, yet the only candidate for the second `"name"` field I have in mind is `INTERNAL_ERROR`.

To be more explicit about the structure we may want to write:

```rust
    InternalError {
        name: String,
        cause: { // This is not a valid Rust syntax, though
            error_message: String 
        }
    },
```

or

```rust
    InternalError(RpcInternalErrorKind),
```